src/sodiumCompatibility/java/net/coderbot/iris/compat/sodium/mixin/shadow_map/MixinSodiumWorldRenderer.java

```diff
@@ -48,7 +49,8 @@
     @Unique
     private void iris$ensureStateSwapped() {
         if (!wasRenderingShadows && ShadowRenderingState.areShadowsCurrentlyBeingRendered()) {
-            if (this.chunkRenderManager instanceof SwappableChunkRenderManager) {
+			ShadowRenderer.visibleBlockEntities.addAll(this.chunkRenderManager.getVisibleBlockEntities());
```

Comment:
Why did you inject here instead of doing something similar to what the 1.17 branch does for Sodium?

Reply:
I wanted to avoid a separate injection, but I can fix that if you want.

Reply:
A separate injection would be fine, this one seems to be an odd spot.

Reply:
Sounds good.